### PR TITLE
implement and fix some methods of android text_input widget

### PR DESCRIPTION
**Pull request by jvmaia**
## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/android/toga_android/widgets/textinput.py [hunk 1]
```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
...
 
     def set_placeholder(self, value):
-        # self.native.cell.placeholderString = self._placeholder
-        self.interface.factory.not_implemented('TextInput.set_placeholder()')
+        self.native.setHint(value)
 
     def set_alignment(self, value):
-        self.interface.factory.not_implemented('TextInput.set_alignment()')
+        self.native.setGravity({
+                LEFT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.LEFT,
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'LEFT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py [hunk 2]
```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
...
+                RIGHT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'RIGHT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py [hunk 3]
```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
...
+                CENTER_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'CENTER_ALIGNED'

File: src/android/toga_android/widgets/textinput.py [hunk 4]
```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
...
+                JUSTIFIED_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'JUSTIFIED_ALIGNED'

File: src/android/toga_android/widgets/textinput.py [hunk 5]
```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
...
+                NATURAL_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'NATURAL_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 1 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'LEFT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 2 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'RIGHT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 3 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'CENTER_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 4 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'JUSTIFIED_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 5 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'NATURAL_ALIGNED'

File: src/android/toga_android/widgets/label.py [hunk 6]
```diff
@@ -18,11 +18,11 @@ def create(self):
...
-                LEFT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.LEFT,
-                RIGHT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
-                CENTER_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                JUSTIFIED_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                NATURAL_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+                'LEFT_ALIGNED': Gravity.CENTER_VERTICAL | Gravity.LEFT,
+                'RIGHT_ALIGNED': Gravity.CENTER_VERTICAL | Gravity.RIGHT,
+                'CENTER_ALIGNED': Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+                'JUSTIFIED_ALIGNED': Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+                'NATURAL_ALIGNED': Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
```
Comment (freakboy3742) [inline src/android/toga_android/widgets/label.py]: This is almost certainly a result of the change to toga constants, introduced in 0.3.0 when style handling was refactored. Alignment constants were modified to follow CSS naming - `from toga.constants import LEFT, RIGHT, CENTER, JUSTIFY`.

File: src/android/toga_android/widgets/textinput.py [hunk 7]
```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
...
+            }[value])
 
     def set_font(self, value):
         self.interface.factory.not_implemented('TextInput.set_font()')
 
     def get_value(self, value):
-        self.interface.factory.not_implemented('TextInput.get_value()')
+        return str(self.native.getText())
```
Comment (freakboy3742) [inline src/android/toga_android/widgets/textinput.py]: Is the str() cast needed here?

File: src/android/toga_android/widgets/label.py (hunk 6 above)
Comment (jvmaia) [reply src/android/toga_android/widgets/label.py]: fixing the import, this changes are only speculative for now.

File: src/android/toga_android/widgets/textinput.py (hunk 7 above)
Comment (jvmaia) [reply src/android/toga_android/widgets/textinput.py]: Yes, if you try to call len(...getText()), without cast str() before, you got some error like: `TypeError: object of type 'org.python.java.Object' has no len()`, you can test it changing the line https://github.com/jvmaia/celsius_fahrenheit-voc/blob/master/celsius_fahrenheit/app.py#L63

File: src/android/toga_android/widgets/label.py [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 from travertino.size import at_least
 
-from toga.constants import *
+from toga.constants import LEFT, RIGH, CENTER, JUSTIFY
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/label.py]: At column 1: (F401) 'toga.constants.RIGH' imported but unused

File: src/android/toga_android/widgets/label.py [hunk 9]
```diff
@@ -2,7 +2,7 @@
...
 
 
 class TogaLabel(extends=android.widget.TextView):
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/label.py]: At column 25: (F821) undefined name 'android'

File: src/android/toga_android/widgets/label.py [hunk 10]
```diff
@@ -18,11 +18,10 @@ def create(self):
 
     def set_alignment(self, value):
         self.native.setGravity({
-                LEFT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.LEFT,
-                RIGHT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
-                CENTER_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                JUSTIFIED_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                NATURAL_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+                LEFT: Gravity.CENTER_VERTICAL | Gravity.LEFT,
+                RIGHT: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/label.py]: At column 17: (F821) undefined name 'RIGHT'

File: src/android/toga_android/widgets/textinput.py (hunk 1 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'LEFT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 2 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'RIGHT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 3 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'CENTER_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 4 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'JUSTIFIED_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 5 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'NATURAL_ALIGNED'

File: src/android/toga_android/widgets/label.py [hunk 11]
```diff
@@ -2,7 +2,7 @@
 
 from travertino.size import at_least
 
-from toga.constants import *
+from toga.constants import LEFT, RIGHT, CENTER, JUSTIFY
 
 
 class TogaLabel(extends=android.widget.TextView):
```
Comment (brutusthebee) [inline src/android/toga_android/widgets/label.py]: At column 25: (F821) undefined name 'android'

File: src/android/toga_android/widgets/textinput.py (hunk 1 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'LEFT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 2 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'RIGHT_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 3 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'CENTER_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 4 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'JUSTIFIED_ALIGNED'

File: src/android/toga_android/widgets/textinput.py (hunk 5 above)
Comment (brutusthebee) [inline src/android/toga_android/widgets/textinput.py]: At column 17: (F821) undefined name 'NATURAL_ALIGNED'

**Comment by phildini** [comment]
Hi there! It looks like this PR might be dead, so we're closing it for now. Feel free to re-open it if you'd like to continue. 😄

**Pull request closed without merging**